### PR TITLE
Preview support for additional archive formats

### PR DESCRIPTION
**Pull request by rupak0577**
As discussed in Zenodo issue [#557](https://github.com/zenodo/zenodo/issues/557).

### CODE REVIEW

**Comment by lnielsen** [comment]
Do you know if this is going to read the entire TAR file? From a quick search it looks to me that you would need to scan through the entire tar-file to extract all the file members. This would mean that in practice we couldn't do this per-request as it could be quite a daunting task (see e.g. 22GB tar-file in https://zenodo.org/record/159158)

We would need some sort of caching/pre-rendering in order to make it work in production, but that also goes for other previewers.

**Comment by lnielsen** [comment]
Closing PR because it's been stale for a long time.

File: invenio_previewer/extensions/tar.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    try:
+        with file.open() as fp, tarfile.open(fileobj=fp) as tf:
+            # Detect filenames encoding.
+            sample = ' '.join(tf.getnames()[:max_files_count])
+            if not isinstance(sample, binary_type):
+                sample = sample.encode('utf-16be')
+            encoding = chardet.detect(sample).get('encoding', 'utf-8')
+            for i, info in enumerate(tf.getmembers()):
+                if i > max_files_count:
+                    raise BufferError('Too many files inside the TAR file.')
```
Comment (ashwinvis) [inline invenio_previewer/extensions/tar.py:30]: Looks like a fine implementation to me. Leaving a review on the code based on what I know of tarfile standard library and Python

The code invokes only `TarFile.getnames()` and `TarfFile.getmembers()` method. These look at the list of the files included (equivalent to `tar tvf` posix / unix command) and does not extract the files. Hope that answers @lnielsen's question.

https://docs.python.org/3.10/library/tarfile.html?highlight=tarfile#tarfile.TarFile.getmembers

**Review by ashwinvis — COMMENTED** [review]
Looks like a fine implementation to me. Leaving a review on the code based on what I know of tarfile standard library and Python

**Pull request closed without merging**